### PR TITLE
Detail widget

### PR DESCRIPTION
**Pull request by wongpratan**

### CODE REVIEW

File: views/ABViewDetailCore.js [hunk 1]
```diff
@@ -74,4 +74,31 @@ module.exports = class ABViewDetailCore extends ABViewContainer {
...
    }
+
+   addFieldToDetail(field, yPosition) {
+      if (field == null) return;
+
+      let newView = field.detailComponent().newInstance(this.application, this);
+      if (newView == null) return;
+
+      // set settings to component
+      newView.settings = newView.settings || {};
```
Comment (nh758) [inline views/ABViewDetailCore.js:85]: `??`

**Review by nh758 — APPROVED** [review]

File: views/ABViewDetailItemCore.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+const ABViewWidget = require("../../platform/views/ABViewWidget");
+
+module.exports = class ABViewDetailItemCore extends ABViewWidget {
+   // constructor(values, application, parent, defaultValues) {
+   //    super(values, application, parent, defaultValues);
+   // }
+
+   detailComponent() {
+      var detailView = null;
```
Comment (guyyoo) [inline views/ABViewDetailItemCore.js:9]: var -> let

File: views/ABViewDetailItemCore.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+      var curr = this;
```
Comment (guyyoo) [inline views/ABViewDetailItemCore.js:11]: var -> let

File: views/ABViewDetailItemCore.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      // [aliasName].[columnName]
+      if (field && this.settings.alias) {
+         field.alias = this.settings.alias;
+      }
+
+      return field;
+   }
+
+   getCurrentData() {
+      var detailCom = this.detailComponent();
```
Comment (guyyoo) [inline views/ABViewDetailItemCore.js:50]: var -> let/const

File: views/ABViewDetailItemCore.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      if (!detailCom) return null;
+
+      var dv = detailCom.datacollection;
```
Comment (guyyoo) [inline views/ABViewDetailItemCore.js:53]: var -> let/const

File: views/ABViewDetailItemCore.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      if (!dv) return null;
+
+      var field = this.field();
```
Comment (guyyoo) [inline views/ABViewDetailItemCore.js:56]: var -> let/const

File: views/ABViewDetailItemCore.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      if (!field) return null;
+
+      var currData = dv.getCursor();
```
Comment (guyyoo) [inline views/ABViewDetailItemCore.js:59]: var -> let/const

**Review by jinzsom — APPROVED** [review]
Very small things and then it is good to go.

File: views/ABViewDetailCore.js [hunk 8]
```diff
@@ -74,4 +74,31 @@ module.exports = class ABViewDetailCore extends ABViewContainer {
...
+      newView.settings.fieldId = field.id;
+      newView.settings.labelWidth =
+         this.settings.labelWidth ||
```
Comment (guyyoo) [inline views/ABViewDetailCore.js:88]: ??

File: views/ABViewDetailItemCore.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+      while (
+         !curr.isRoot() &&
+         curr.parent &&
+         curr.key != "detail" &&
+         curr.key != "dataview"
+      ) {
+         curr = curr.parent;
+      }
+
+      if (curr.key == "detail" || curr.key == "dataview") {
```
Comment (guyyoo) [inline views/ABViewDetailItemCore.js:21]: ??

**Review by jinzsom — APPROVED** [review]

**Pull request merged**